DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UIDatePickerRecorder.swift

```diff
@@ -73,11 +73,7 @@ private struct WheelsStyleDatePickerRecorder {
         nodeRecorders: [
             UIPickerViewRecorder(
                 textObfuscator: { context in
-                    switch context.recorder.privacy {
-                    case .allowAll:         return context.textObfuscators.nop
-                    case .maskAll:          return context.textObfuscators.spacePreservingMask
-                    case .maskUserInput:    return context.textObfuscators.spacePreservingMask
-                    }
+                    return context.textObfuscation.staticTextObfuscator(for: context.recorder.privacy)
```

Comment:
💡 Passing `context.*.privacy` to `context.textObfuscation.*()` might look odd for a moment, but it will get more sense once per-element masking is available later in fine grained privacy project:
```swift
context.textObfuscation.staticTextObfuscator(for: overridenPrivacy ?? context.recorder.privacy)
```

Reply:
I think, we don't have a right abstraction here. I kind of liked what Max proposed but with slightly protocol oriented.

```swift
protocol TextObfuscator {
    func obfuscate(for privacyLevel: SessionReplayPrivacy) -> TextObfuscating
}
```

```swift
struct sensitiveTextObfuscator: TextObfuscator {
    func obfuscate(for privacyLevel: SessionReplayPrivacy) -> TextObfuscating {
        ...
    }
}
```

Usage

```swift
context.textObfuscator.obfuscate(for: context.recorder.privacy)
```

Reply:
@maxep there is the performance characteristics we anticipate in `SpacePreservingMaskObfuscator` that could eventually make this idea slightly less convenient. It works `O(n)` today, but `O(1)` is possible with LRU-like caching. It runs on background thread, so it's not that much of a problem, but anticipating apps with long texts (e.g. social media posts), it could make sense to cache it. If adding caching, it the `context.textObfuscator` would become truer dependency injection mechanism.

However, thinking twice I tend to realise that it might be pre-mature. Adding cache to `SPMObfuscator` is not impossible with your proposal, so I updated code with what you proposed. I agree that it results with more consistent domain definition 👌.

---

@ganeshnj if I get the idea right, this won't be correct to our problematic:

```swift
context.textObfuscator.obfuscate(for: context.recorder.privacy)
```

Some cases are more complex as this, e.g. text field masking is currently:

```swift
if isPlaceholder {
    return context.recorder.privacy.hintTextObfuscator
} else if isSensitive {
    return context.recorder.privacy.sensitiveTextObfuscator
} else {
    return context.recorder.privacy.inputAndOptionTextObfuscator
}
```

so it can't use single `context.textObfuscator` with different implementations. There is also a challenge of elements that need to be masked differently in different subtrees, described in more details in #1237 (2nd point in "How?").

I think we're "good enough" with @maxep 's proposal.

Reply:
for my understanding

are `isPlaceholder`, `isSensitive` set before the obfuscator is decided or later?

Reply:
They are set before (we infer both from `UITextField` instance). Based on that, we decide if the text should be classified as "Hint Text", "Sensitive Text" or "Input & Option Text" according to [Privacy Options in Mobile SR](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2945942237/Privacy+Options+in+Mobile+Session+Replay). The category of the text and current `privacy` are then used to decide on which obfuscator implementation will be used: `NOP`, `FixLength` or `SpacePreserving`.